### PR TITLE
Introduce dedicated JAXP extension 

### LINKED ISSUE #12129: JAXP extension
**Describe the extension**

The current JAXB extension contains some JAXP stuff:

* https://github.com/quarkusio/quarkus/blob/1.8.0.Final/extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java#L196-L199
* https://github.com/quarkusio/quarkus/blob/1.8.0.Final/extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java#L205-L206
* https://github.com/quarkusio/quarkus/blob/1.8.0.Final/extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java#L210-L214

It would be handy to have separate JAXP-only extension

Interested in this extension, please +1 via the emoji/reaction feature of GitHub (top right).

**Configuration suggestion**

None

**Additional context**

https://github.com/apache/camel-quarkus/issues/1796

### PR DESCRIPTION
**Pull request by ppalaga**
Fix #12129

### CODE REVIEW

File: integration-tests/pom.xml [hunk 1]
```diff
@@ -136,6 +136,7 @@
         <module>micrometer-mp-metrics</module>
         <module>micrometer-prometheus</module>
         <module>logging-json</module>
+        <module>jaxp</module>
```
Comment (gsmet) [inline integration-tests/pom.xml:139]: This module needs to be added to `ci-actions.yml` to be included in the native test workflow. Add it to one of the Misc category (the shortest one and add 5 minutes to the timeout).

File: extensions/jaxp/runtime/src/main/resources/META-INF/quarkus-extension.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+name: "JAXP"
+metadata:
+  categories:
+  - "xml"
```
Comment (gsmet) [inline extensions/jaxp/runtime/src/main/resources/META-INF/quarkus-extension.yaml:5]: I wonder if we should make it unlisted or not? It looks like a very technical one?

**Review by gsmet — CHANGES_REQUESTED** [review]
Thanks! Looks perfect except for CI and I added a question about the descriptor.

File: extensions/jaxp/runtime/src/main/resources/META-INF/quarkus-extension.yaml (hunk 2 above)
Comment (geoand) [reply extensions/jaxp/runtime/src/main/resources/META-INF/quarkus-extension.yaml:5]: +1 for making it unlisted

File: integration-tests/jaxp/pom.xml [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-jaxp-deployment</artifactId>
```
Comment (geoand) [inline integration-tests/jaxp/pom.xml:54]: Is this really needed?

File: extensions/jaxp/runtime/src/main/resources/META-INF/quarkus-extension.yaml (hunk 2 above)
Comment (ppalaga) [reply extensions/jaxp/runtime/src/main/resources/META-INF/quarkus-extension.yaml:5]: I think that there may well exist end users needing DocumentBuilder and XPath, but I have no problem to unlist if you insist on that.

File: integration-tests/jaxp/pom.xml (hunk 3 above)
Comment (ppalaga) [reply integration-tests/jaxp/pom.xml:54]: It is for for consistent build order. See https://github.com/quarkusio/quarkus/pull/11331 for more details.

Comment (geoand) [reply integration-tests/jaxp/pom.xml:54]: Understood, thanks

File: integration-tests/pom.xml (hunk 1 above)
Comment (ppalaga) [reply integration-tests/pom.xml:139]: Done in f3e6f14df24b908373f09a8defa29a424aaa0d0e

File: extensions/jaxp/runtime/src/main/resources/META-INF/quarkus-extension.yaml (hunk 2 above)
Comment (ppalaga) [reply extensions/jaxp/runtime/src/main/resources/META-INF/quarkus-extension.yaml:5]: unlisted in f3e6f14df24b908373f09a8defa29a424aaa0d0e

**Comment by ppalaga** [comment]
f3e6f14df24b908373f09a8defa29a424aaa0d0e: resoved all requests and questions

**Review by geoand — APPROVED** [review]

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
I adjusted the metadata a bit as we are very shy in adding new categories. `serialization` is not perfect for sure but we try to avoid having too many categories and as it's unlisted it's not that big of a deal (I wasn't sure how categories for unlisted extensions would be properly filtered or not...).

**Pull request merged**